application-analytics-ui/src/main/resources/Analytics/Code/Macros/VisitsSummary.xml

```diff
@@ -220,45 +228,46 @@ require(['jquery', 'chart'], function ($) {
   }
 
   // Updates the chart.
-  function editCanvas(canvas, cacheId, date, endpoints, fields, period, yAxeNames) {
+  function editCanvas(canvas, cacheId, date, endpoints, fields, period, axis) {
     let requestPageRef = XWiki.Model.resolve('Analytics.Code.Macros.VisitsSummaryJson', XWiki.EntityType.DOCUMENT);
     let url = new XWiki.Document(requestPageRef).getURL('get');
-    let requests = []; // This is correctly defined
-    // A field can have multiple endpoints(lines) call all of them and store the promise in vector for later use.
-    for (let i = 0; i &lt; endpoints.length; i++) {
+    let requestsPromises = [];
+    // A field can have multiple endpoints(lines in the graph eg Unique Views and Views) call all of them and store the promise in vector for later use.
```

Comment:
`// A field can have multiple endpoints (lines in the graph, e.g. Unique Views and Views). Call all of them and store the promise in vector for later use.`